src/wazuh/client.ts

```
@@ -56,37 +55,97 @@ class WazuhClient {
       },
       agent,
     };
+    // https://documentation.wazuh.com/current/user-manual/api/reference.html#section/
+    // jwt expires every 900 seconds
+    this.refreshAuthInterval = setInterval(async () => {
+      await makeRequest(`${this.config.managerUrl}/security/config`, {
+        ...this.requestOptions,
+        method: 'POST',
+        body: JSON.stringify({
+          auth_token_exp_timeout: 900,
+        }),
+      });
+    }, 800000);
+
+    this.initialized = true;
   }
 
   public destroy() {
     clearInterval(this.refreshAuthInterval);
+    this.initialized = false;
   }
 
   public async verifyAccess() {
     return this.fetchData('/manager/info');
   }
 
   public async fetchManager(): Promise<WazuhManager> {
-    return this.fetchData<WazuhManager>('/manager/info');
+    const items = await this.fetchData<WazuhManager>('/manager/info');
+    if (items.length === 1) {
+      return items[0];
+    } else {
+      if (items.length) {
+        this.logger.warn(
+          {
+            managerInfo: items.map((item) => {
+              return {
+                type: item.type,
+                version: item.version,
+              };
+            }),
+          },
+          `Wazuh Api responded back with ${items.length} Managers when we only expected one. We only consumed the first one.`,
+        );
+        return items[0];
+      } else {
+        throw new IntegrationProviderAPIError({
+          endpoint: `${this.config.managerUrl}/manager/info`,
+          status: '500',
+          statusText: 'Internal Error',
+          message:
+            'empty non-error response from Wazuh API when fetching manager info',
+        });
```

Should we just throw the error always? If we don't expect it and then it happens we should probably be alerted in some way.

---

I'm not sure honestly, perhaps this is a good use case for `publishWarnEvent`, but we'd need to make a change to the sdk to add a new reason. Something like `Unexpected Server Response`.
I am hoping to get this fix out asap though so I'd love to push this to a small follow-up ticket.